portal-impl/src/com/liferay/portal/service/impl/UserServiceImpl.java

```diff
@@ -3528,8 +3528,13 @@ protected long[] checkOrganizations(long userId, long[] organizationIds)
 
 		PermissionChecker permissionChecker = getPermissionChecker();
 
-		boolean strictAssignment = PropsValues.ORGANIZATIONS_ASSIGNMENT_STRICT ||
-								   !permissionChecker.isCompanyAdmin();
+		boolean strictAssignment = false;
+
+		if (PropsValues.ORGANIZATIONS_ASSIGNMENT_STRICT ||
+			!permissionChecker.isCompanyAdmin()) {
+
+			strictAssignment = true;
+		}
```

Comment:
why here is done with the if and not like in the previous commit with = conditions?
please have them consistent

Reply:
and move it before the loop where you use it